api/src/main/java/org/eclipse/microprofile/metrics/MetadataBuilder.java

```diff
@@ -193,15 +193,19 @@ public Metadata build() {
         switch (type) {
             case HISTOGRAM:
             case GAUGE:
-            if(Objects.isNull(unit) || MetricUnits.NONE.equals(unit)) {
-                throw new IllegalStateException("A unit is required for this type of metric");
-            }
+                if (Objects.isNull(unit) || MetricUnits.NONE.equals(unit)) {
+                    throw new IllegalStateException("A unit is required for this type of metric");
+                }
+                break;
             case TIMER:
             case METERED:
             case COUNTER:
-                if(Objects.nonNull(unit) || !MetricUnits.NONE.equals(unit)) {
+                if (Objects.nonNull(unit) || !MetricUnits.NONE.equals(unit)) {
```

Comment:
I'm not sure what it would mean for a Timer or Metered to have a unit -- from end user perspective putting a unit on one of those wouldn't have much intuitive meaning.  For a counter, I can see @pilhuhn 's point.

Also, is the if condition incorrectly specified?  I think it should be
`if (Objects.nonNull(unit) && !MetricUnits.NONE.equals(unit))`

Reply:
That said, a counter is implicitly always counting "instances", isn't it?  I'm not sure I have a strong opinion either way on this.  

(that said, please don't forget my comment above about the if condition!)

Reply:
Also needs a spec update + changelog